ports/stm32/boards/NUCLEO_WB55/rfcore_firmware.py

```diff
@@ -0,0 +1,545 @@
+        elif _STATE_id == _STATE_CHECK_UPDATES:
+            log("Checking for updates")
+            fus_version = stm.rfcore_fw_version(_FW_VERSION_FUS)
+            log("FUS version {}", fus_version)
+            if fus_version < _FUS_VERSION_102:
+                log("Factory FUS detected")
+                if _stat_and_start_copy(
+                    _PATH_FUS_102, _ADDR_FUS, _STATE_COPYING_FUS, _STATE_COPIED_FUS
+                ):
+                    continue
+            elif fus_version >= _FUS_VERSION_102 and fus_version < _FUS_VERSION_110:
+                log("FUS 1.0.2 detected")
+                if _stat_and_start_copy(
+                    _PATH_FUS_110, _ADDR_FUS, _STATE_COPYING_FUS, _STATE_COPIED_FUS
+                ):
+                    continue
+            else:
+                log("FUS is up-to-date")
+
+            if fus_version >= _FUS_VERSION_110:
+                if _stat_and_start_copy(
+                    _PATH_WS_BLE_HCI, _ADDR_WS_BLE_HCI, _STATE_COPYING_WS, _STATE_COPIED_WS
+                ):
+                    continue
+                else:
+                    log("No WS updates available")
+            else:
+                # Don't attempt to install WS if we're running an old FUS.
+                log("Need latest FUS to install WS")
+
+            # Attempt to go back to WS.
+            # Either this will fail (because WS was removed due to FUS install), or
+            # this whole thing was a no-op and we should be fine to restart WS.
+            _write_state(_STATE_WAITING_FOR_WS)
+
+        # This shouldn't happen - the flash write should always complete and
+        # move straight onto the COPIED state. Failure here indicates that
+        # the rfcore is misconfigured or the WS firmware was not deleted first.
+        elif _STATE_id == _STATE_COPYING_FUS or _STATE_id == _STATE_COPYING_WS:
+            log("Flash copy failed mid-write")
+            _write_failure_state(REASON_FLASH_COPY_FAILED)
+
+        # Flash write completed, we should immediately see GET_STATE return 0,0
+        # so we can start the FUS install.
+        elif _STATE_id == _STATE_COPIED_FUS:
+            if fus_is_idle():
+                log("FUS copy complete, installing")
+                _write_state(_STATE_INSTALLING_FUS)
+                _fus_run_fwupgrade(_ADDR_FUS)
+            else:
+                log("FUS copy bad state")
+                _write_failure_state(REASON_FLASH_FUS_BAD_STATE)
+
+        # Keep polling the state until we see a 0,0 (success) or non-transient
+        # error. In general we should expect to see (16,0) several times,
+        # followed by a (255,0), followed by (0, 0).
+        elif _STATE_id == _STATE_INSTALLING_FUS:
+            log("Installing FUS...")
+            status, result = fus_get_state()
+            log("FUS state: {} {}", status, result)
+            if status >= 0x20 and status <= 0x2F and result == 0:
```

Comment:
pylint suggested changing this to `if 0x20 <= status <= 0x2F and result == 0:`... maybe a good idea for readability?